### ISSUE TITLE
Terraform 1.8.x builtin functions break tflint syntax parsing

### ISSUE THREAD

**Issue by lonelyelk** [terraform-linters/tflint#2028]
### Summary

Terraform code with [builtin functions](https://developer.hashicorp.com/terraform/language/functions/terraform-decode_tfvars) (e.g. `provider::terraform::decode_tfvars` or `provider::terraform::encode_tfvars`) will always fail in tflint as if it was incorrect code without specific linter rule. At the same time code is correct and running.

For example:

```terraform
terraform {
  required_providers {
    terraform = {
      source = "terraform.io/builtin/terraform"
    }
  }
  required_version = "~> 1.8"
}

locals {
  test = provider::terraform::decode_tfvars(
    <<EOT
      example = "Hello!"
    EOT
  )
}

output "example" {
  value = local.test["example"]
}
```

Gives:

```text
tflint.tf:10,18-19: Missing newline after argument; An argument definition must end with a newline.:

Error: Missing newline after argument

  on tflint.tf line 10, in locals:
  10:   test = provider::terraform::decode_tfvars(

An argument definition must end with a newline.
```

And another one:

```terraform
terraform {
  required_providers {
    terraform = {
      source = "terraform.io/builtin/terraform"
    }
  }
  required_version = "~> 1.8"
}

locals {
  test = lookup(
    provider::terraform::decode_tfvars(
      <<EOT
        example = "Hello!"
      EOT
    ),
    "example",
    "default"
  )
}

output "example" {
  value = local.test
}
```

Gives:

```text
tflint.tf:11,13-14: Missing argument separator; A comma is required to separate each function argument from the next.:

Error: Missing argument separator

  on tflint.tf line 11, in locals:
  10:   test = lookup(
  11:     provider::terraform::decode_tfvars(

A comma is required to separate each function argument from the next.
```

The underscode in both errors is under the first ":" character after `provider` namespace.

### Command

tflint

### Terraform Configuration

```terraform
terraform {
  required_providers {
    terraform = {
      source = "terraform.io/builtin/terraform"
    }
  }
  required_version = "~> 1.8"
}
```

### TFLint Configuration

```HCL
any
```

### Output

```Shell
tflint.tf:10,18-19: Missing newline after argument; An argument definition must end with a newline.:

Error: Missing newline after argument

  on tflint.tf line 10, in locals:
  10:   test = provider::terraform::decode_tfvars(

An argument definition must end with a newline.


--------


tflint.tf:11,13-14: Missing argument separator; A comma is required to separate each function argument from the next.:

Error: Missing argument separator

  on tflint.tf line 11, in locals:
  10:   test = lookup(
  11:     provider::terraform::decode_tfvars(

A comma is required to separate each function argument from the next.
```

### TFLint Version

0.50.3

### Terraform Version

1.8.x

### Operating System

- [X] Linux
- [X] macOS
- [X] Windows

**Comment by lonelyelk** [comment]
In both cases `terraform fmt .` and `terraform apply` run without problems
